pantheon-decoupled.php

```diff
@@ -448,6 +467,128 @@ function pantheon_decoupled_redirect_to_fes() {
 	}
 }
 
+function pantheon_decoupled_regen_env_vars() {
+  if ( ! current_user_can( 'manage_options' ) ) {
+    wp_die( esc_html__( 'You do not have sufficient permissions to access this page.', 'wp-decoupled-preview' ) );
+  }
+  check_admin_referer( 'env-regen', 'nonce' );
+
+  // Get data for preview site
+  $id = isset( $_GET['id'] ) ? absint( sanitize_text_field( $_GET['id'] ) ) : NULL;
+  $preview_sites = get_option( 'preview_sites' );
+  $preview_site = isset( $preview_sites['preview'][ $id ] ) ? $preview_sites['preview'][ $id ] : NULL;
+  
+  $docs_url = 'https://docs.pantheon.io/guides/decoupled/overview'
+    
+  ?>
+      <style>
+        /* Hide admin bar and padding on top of page. */
+        html.wp-toolbar {
+          padding-top: 0;
+        }
+        #wpadminbar {
+          display: none;
+        }
+      </style>
+      <div class="wrap">
+          <h1><?php esc_html_e( 'Regenerate Environment Variables', 'wp-pantheon-decoupled' ); ?></h1>
+          <?php
+        if ( $id ) {
+          $site_label = $preview_site['label'];
+          $url = wp_nonce_url( add_query_arg( [
+            'page' => 'env_regen_action',
+            'id' => $id,
+          ], admin_url( 'options-general.php' ) ), 'env-regen', 'nonce' );
+          ?>
+          <a id="regen-password" class="button-secondary button-large" href="<?php echo esc_url( $url ); ?>">
+            <?php
+            echo esc_html(
+              // Translators: %s is the preview site label.
+              sprintf( __( 'Regenerate %s WP_APPLICATION_PASSWORD', 'wp-decoupled-preview' ), $site_label )
+            );
+            ?>
+          </a>
+          <p>Checkout the Pantheon
+            <a id="docs-link" target="_blank" href="<?php echo esc_url( $docs_url ); ?>">
+              platform docs
+            </a> for more information about Front-End Site configuration.
+          </p>
+          <?php
+        }
+        ?>
+      </div>
+  <?php
+}
+
+/**
+ * Helper to get password by name.
+ *
+ * @return string
+ */
+function pantheon_decoupled_get_user_application_password( $user_id, $id ) {
```

Comment:
The [get_user_application_password()](https://developer.wordpress.org/reference/classes/wp_application_passwords/get_user_application_password/) function in `WP_Application_Passwords` requires a passwords `UUID` . I made a copy of this function which accepts `id` as an argument instead to make password fetching more straightforward.